app/src/main/java/br/com/catbag/gifreduxsample/ui/components/GifsAdapter.java

```diff
@@ -0,0 +1,64 @@
+package br.com.catbag.gifreduxsample.ui.components;
+
+import android.support.v7.widget.RecyclerView;
+
+import com.google.common.collect.ImmutableMap;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+
+import br.com.catbag.gifreduxsample.models.Gif;
+import trikita.anvil.RenderableRecyclerViewAdapter;
+
+/**
+ * Created by raulcca on 11/18/16.
+ */
+
+public final class GifsAdapter extends RenderableRecyclerViewAdapter {
+    private GifsAdapterRenderable mRenderable;
+    private Map<String, Gif> mGifs;
+    private List mUuids;
+
+    private GifsAdapter(Map<String, Gif> gifs, GifsAdapterRenderable renderable) {
+        setGifs(gifs);
+        mRenderable = renderable;
+    }
+
+    public static GifsAdapter gifsAdapter(Map<String, Gif> gifs, GifsAdapterRenderable renderable) {
+        GifsAdapter adapter = new GifsAdapter(gifs, renderable);
+        adapter.setHasStableIds(false);
+        return adapter;
+    }
+
+    public Map getGifs() {
+        return mGifs;
+    }
+
+    public void setGifs(Map<String, Gif> gifs) {
+        mGifs = ImmutableMap.copyOf(gifs);
```

Comment:
Firstly, I tried to resolve that problem of two activities using the same state on adapter, but now thinking with more attention, it isn't the solution... :(, the value can be changed on exact time of using inside view...the unique solution maybe a synchronized one. You have this opinion too?